crates/miden-objects/src/account/component/metadata/storage/toml/tests.rs

```diff
@@ -0,0 +1,714 @@
+
+    let typed_value = SchemaTypeIdentifier::native_word();
+    assert_eq!(value_slot.word(), &WordSchema::new_singular(typed_value.clone()));
+
+    let map_slot = schema
+        .fields()
+        .get(&StorageSlotName::new("demo::typed_map").unwrap())
+        .expect("map slot missing");
+    let map_slot = match map_slot {
+        StorageSlotSchema::Map(slot) => slot,
+        _ => panic!("expected map slot"),
+    };
+
+    let pub_key_type = SchemaTypeIdentifier::new("auth::rpo_falcon512::pub_key").unwrap();
+    assert_eq!(map_slot.key_schema(), &WordSchema::new_singular(pub_key_type.clone()));
+    assert_eq!(map_slot.value_schema(), &WordSchema::new_singular(pub_key_type));
+
+    let mut requirements = metadata.init_value_requirements();
+    assert_eq!(
+        requirements
+            .remove(&"demo::typed_value".parse::<StorageValueName>().unwrap())
+            .unwrap()
+            .r#type,
+        typed_value
+    );
+    assert!(!requirements.contains_key(&"demo::typed_map".parse::<StorageValueName>().unwrap()));
+
+    let round_trip = metadata.to_toml().expect("serialize");
+    let parsed: toml::Value = toml::from_str(&round_trip).unwrap();
+    let storage = parsed.get("storage").unwrap().as_table().unwrap();
+    let storage_values = storage.get("value").unwrap().as_array().unwrap();
+    let storage_maps = storage.get("map").unwrap().as_array().unwrap();
+
+    let typed_value_entry = storage_values
+        .iter()
+        .find(|entry| entry.get("name").unwrap().as_str().unwrap() == "demo::typed_value")
+        .unwrap();
+    assert_eq!(typed_value_entry.get("type").unwrap().as_str().unwrap(), "word");
+
+    let typed_map_entry = storage_maps
+        .iter()
+        .find(|entry| entry.get("name").unwrap().as_str().unwrap() == "demo::typed_map")
+        .unwrap();
+    assert_eq!(
+        typed_map_entry.get("key-type").unwrap().as_str().unwrap(),
+        "auth::rpo_falcon512::pub_key"
+    );
+    assert_eq!(
+        typed_map_entry.get("value-type").unwrap().as_str().unwrap(),
+        "auth::rpo_falcon512::pub_key"
+    );
+}
+
+#[test]
+fn extensive_schema_metadata_and_init_toml_example() {
+    let metadata_toml = r#"
+        name = "Extensive Example"
+        description = "Exercises composed slots, singular typed slots, static maps, optional init maps, and map typing."
+        version = "0.1.0"
+        supported-types = ["FungibleFaucet", "RegularAccountImmutableCode"]
+
+        # composed slot schema expressed via `type = [...]`
+        [[storage.value]]
+        name = "demo::token_metadata"
+        description = "Token metadata: max_supply, symbol, decimals, reserved."
+        type = [
+            { type = "u32", name = "max_supply", description = "Maximum supply (base units)" },
+            { type = "token_symbol", name = "symbol", default-value = "TST" },
+            { type = "u8", name = "decimals", description = "Token decimals" },
+            { type = "void" }
+        ]
```

Comment:
What happens if `type` array contains less than four entries?

Reply:
It will error as it expects exactly 4. As you suggested in the other comment, we could make `void` the default type and have it fill the remaining elements.